src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -386,6 +403,76 @@ private static String InternalReadAllText(String path, Encoding encoding)
                 return sr.ReadToEnd();
         }
 
+
+        // If we use the path-taking constructors we will not have FileOptions.Asynchronous set and
+        // we will have asynchronous file access faked by the thread pool. We want the real thing.
+        private static StreamReader AsyncStreamReader(string path, Encoding encoding)
+        {
+            FileStream stream = new FileStream(
+                path, FileMode.Open, FileAccess.Read, FileShare.Read, DefaultBufferSize,
+                FileOptions.Asynchronous | FileOptions.SequentialScan);
+
+            return new StreamReader(stream, encoding, true);
+        }
+
+        private static StreamWriter AsyncStreamWriter(string path, bool append, Encoding encoding)
+        {
+            FileStream stream = new FileStream(
+                path, append ? FileMode.Append : FileMode.Create, FileAccess.Write, FileShare.Read, DefaultBufferSize,
+                FileOptions.Asynchronous | FileOptions.SequentialScan);
+
+            return new StreamWriter(stream, encoding);
+        }
+
+        public static Task<string> ReadAllTextAsync(
+            string path, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return ReadAllTextAsync(path, Encoding.UTF8, cancellationToken);
+        }
+
+        [System.Security.SecuritySafeCritical] // auto-generated
+        public static Task<string> ReadAllTextAsync(
+            string path, Encoding encoding, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+            if (path.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
+
+            Contract.EndContractBlock();
```

Comment:
Likewise, I understand these have no effect for us, but I put them in analogous places to the non-async code for now.